RIOT-OS/RIOT
# sys/random/kconfig: remove HWRNG default in backend choice

### Contribution description

The discrepancy between the default backend choice for random in Kconfig and Makefile requires to add an extra configuration line in many applications. Although usually it is preferred to use the Hardware RNG over a software one, for now we match the Makefile.dep default choice, until migration is complete. 

### Testing procedure
- Green murdock
- Check that the comment suggesting to use the hardware RNG shows up properly in `menuconfig`.

### Issues/PRs references
#18024
